### PR TITLE
ref(modal): add backdropclick to modal

### PR DESCRIPTION
**Pull request by JeanBeaurepaire**
## Description

Add an event when there is a click on the modal backdrop.

-----

### CODE REVIEW

**Comment by LuccaIntegration** [comment]
:woman_cook: http://lucca-front.lucca.local/PR-2044/storybook

File: packages/ng/popup/src/lib/popup-ref.model.ts [hunk 1]
```diff
@@ -131,14 +134,18 @@ export abstract class ALuPopupRef<T = unknown, D = unknown, R = unknown, C exten
...
-			const bdClicked$ = this._overlayRef.backdropClick();
-			const escPressed$ = this._overlayRef.keydownEvents().pipe(filter((evt) => evt.key === 'Escape'));
-			const sub = merge(bdClicked$, escPressed$)
-				.pipe(first())
-				.subscribe((_e) => this.dismiss());
-			this._subs.add(sub);
-		}
+		const bdClicked$ = this._overlayRef.backdropClick();
+		const escPressed$ = this._overlayRef.keydownEvents().pipe(filter(({ key }) => key === 'Escape'));
+		const escPressedSub = escPressed$.pipe(filter(() => !this._config.undismissable)).subscribe((_e) => this.dismiss());
```
Comment (GuillaumeNury) [inline packages/ng/popup/src/lib/popup-ref.model.ts:139]:
```suggestion
    if ( !this._config.undismissable) {
		  this._subs.add(escPressed$.subscribe(() => this.dismiss()));
		}
```

File: packages/ng/popup/src/lib/popup-ref.model.ts [hunk 2]
```diff
@@ -131,14 +134,18 @@ export abstract class ALuPopupRef<T = unknown, D = unknown, R = unknown, C exten
...
+		const bdClickedSub = bdClicked$.pipe().subscribe((_e) => {
+			this.onBackdropClick.next();
+			if (!this._config.undismissable) {
+				this.onBackdropClick.complete();
```
Comment (GuillaumeNury) [inline packages/ng/popup/src/lib/popup-ref.model.ts:143]: We could complete it in `_completeSubjects` method

**Comment by LuccaIntegration** [comment]
:woman_cook: http://lucca-front.lucca.local/PR-2044/storybook

**Review by GuillaumeNury — APPROVED** [review]

**Review by jeremie-lucca — APPROVED** [review]

**Pull request merged**